### PR TITLE
Add event checks to unit tests

### PR DESCRIPTION
**Pull request by ChaoticWalrus**
This PR is still a bit of a WIP.
It is OK to merge (all tests currently passing, pure improvement), but it still needs coverage added to the StrategyBase tests, at least.

### CODE REVIEW

File: src/test/EigenPod.t.sol [hunk 1]
```diff
@@ -58,10 +58,49 @@ contract EigenPodTests is ProofParsing, EigenPodPausingConstants {
     bytes32[] withdrawalFields;
     bytes32[] validatorFields;
 
+
+    // EIGENPODMANAGER EVENTS
+    /// @notice Emitted to notify the update of the beaconChainOracle address
+    event BeaconOracleUpdated(address indexed newOracleAddress);
```
Comment (gpsanant) [inline src/test/EigenPod.t.sol:64]: Is it possible to import the events like in the examples here: https://book.getfoundry.sh/cheatcodes/expect-emit

File: src/test/unit/DelayedWithdrawalRouterUnit.t.sol [hunk 2]
```diff
@@ -131,7 +143,13 @@ contract DelayedWithdrawalRouterUnitTests is Test {
         delayedWithdrawalRouter.createDelayedWithdrawal{value: delayedWithdrawalAmount}(podOwner, address(0));
     }
 
-    function testClaimDelayedWithdrawals(uint8 delayedWithdrawalsToCreate, uint8 maxNumberOfDelayedWithdrawalsToClaim, uint256 pseudorandomNumber_, address recipient, bool useOverloadedFunction)
+    function testClaimDelayedWithdrawals(
+        uint8 delayedWithdrawalsToCreate,
+        uint8 maxNumberOfDelayedWithdrawalsToClaim,
+        uint256 pseudorandomNumber_,
```
Comment (gpsanant) [inline src/test/unit/DelayedWithdrawalRouterUnit.t.sol:149]: looked into the file and doesn't seem like we use this param?

Comment (gpsanant) [reply src/test/unit/DelayedWithdrawalRouterUnit.t.sol:149]: looks like we use the _getPseudorandomNumber function though

File: src/test/EigenPod.t.sol [hunk 3]
```diff
@@ -614,6 +664,8 @@ contract EigenPodTests is ProofParsing, EigenPodPausingConstants {
         eigenPodManager.stake{value: 12 ether}(_pubkey, _signature, _depositDataRoot);
         
         // successful call
+        cheats.expectEmit(true, false, false, false);
```
Comment (gpsanant) [inline src/test/EigenPod.t.sol:667]: no address check?

File: src/test/unit/StrategyManagerUnit.t.sol [hunk 4]
```diff
@@ -463,6 +525,10 @@ contract StrategyManagerUnitTests is Test, Utils {
         cheats.startPrank(strategyManager.owner());
         IStrategy[] memory _strategy = new IStrategy[](1);
         _strategy[0] = IStrategy(address(reenterer));
+        for (uint256 i = 0; i < _strategy.length; ++i) {
```
Comment (gpsanant) [inline src/test/unit/StrategyManagerUnit.t.sol:528]: these 5 lines are used a lot, feel like we could make them another function

File: src/test/EigenPod.t.sol (hunk 3 above)
Comment (ChaoticWalrus) [reply src/test/EigenPod.t.sol:667]: pretty sure I just copy-pasted the existing call to `cheats.expectEmit`. I'll fix it everywhere

File: src/test/unit/DelayedWithdrawalRouterUnit.t.sol (hunk 2 above)
Comment (ChaoticWalrus) [reply src/test/unit/DelayedWithdrawalRouterUnit.t.sol:149]: we are using indeed using this input. we set the storage `_pseudorandomNumber` equal to the `pseudorandomNumber_`. The storage value is then used inside the `_getPseudorandomNumber` function.

File: src/test/EigenPod.t.sol (hunk 1 above)
Comment (ChaoticWalrus) [reply src/test/EigenPod.t.sol:64]: can't sort this out, not sure it's possible how you're picturing -- the foundry docs are unclear to me.
if you can sort this out then go right ahead.

File: src/test/EigenPod.t.sol (hunk 3 above)
Comment (ChaoticWalrus) [reply src/test/EigenPod.t.sol]: fixed in https://github.com/Layr-Labs/eigenlayer-contracts/pull/44/commits/b9be2206e32fd06a43963152bc5e128eb6138a17

File: src/test/unit/StrategyManagerUnit.t.sol (hunk 4 above)
Comment (ChaoticWalrus) [reply src/test/unit/StrategyManagerUnit.t.sol:528]: we use this 4 times. I think it is OK to not make a function for now.

File: src/test/EigenPod.t.sol (hunk 1 above)
Comment (gpsanant) [reply src/test/EigenPod.t.sol:64]: not a big deal, just will error out if we forget to propagate.

File: src/test/unit/StrategyManagerUnit.t.sol (hunk 4 above)
Comment (gpsanant) [reply src/test/unit/StrategyManagerUnit.t.sol:528]: sounds good

**Review by gpsanant — APPROVED** [review]
thanks for adding this, huge for our subgraph correctness

**Comment by ChaoticWalrus** [comment]
Leaving the branch open for the moment, so I can add more coverage and open another PR.

**Pull request merged**